scripts/2025-10-22-Spike-migration-templates/README.md

```diff
@@ -0,0 +1,207 @@
+# Spike migration to using templates
+
+At this stage of development we have multiple bricks ready to be used:
+- An endpoint that takes a recipe and transform it into a recipe template
+- An endpoint that takes a recipe template and verifies whether it is valid or not
+- A multiplatform library (node, ios, android) that can take a recipe template and render it into a recipe
+
+The goal of this spike is to evaluate how close we are to triggereing the migration.
+
+We'll read recipes from CAPI and transform them into recipe templates.
+
+This will inevitably raise all the recipe templates that aren't considered to be valid by the validation endpoint.
+
+We'll then review invalid recipe templates and decide whether we fix the checking logic, the template generation logic, or accept the differences.
+
+## Diary
+
+### 2025-10-22
+- initialise repo and readme
+- write code that load the index json, load the recipe from CAPI, then find the recipe data within the article
+- improve the comparison logic to handle small and fixable differences, such quotes, whitespace etc
+
+### 2025-10-23
+- continue to iterate on the script
+- have experienced prompt instability, meaning the slightest change in my prompt leads to wildly different results
+  - have linked some of the instability and inconsistencies to the fact sonnet seems to not have apostrophe `’` and double quotes `“”` token output
+  - this means when it try to output these because they are in the original text, it instead outputs `'` and `""` which then leads to invalid json syntax
+  - I've worked around this by cleaning up the input to the model, replacing all instances of `’` with `'` and `“”` with `""`.
+  - I must admit I'm pretty chuffed I found that because I was starting to get worried.
+  => there's a general problem though, which is that single prompt is asking a lot for a model the size of sonnet 4.5, and long term we probably want to chunk that processing in multiple prompts.
+- sonnet 4.5 is having availability issues (3pm, which is start of the day for the east coast), so I've added a 503 return code handler with a 10s sleep. That's not enough, and this is seriously slowing me down.
```

Comment:
Hmmm we could use inference profiles and/or manually spread across supported regions to help with throughput here